swan-lake/development-tutorials/observability/supported-observability-tools-and-platforms/prometheus.md

```diff
@@ -0,0 +1,164 @@
+```
+
+To support Prometheus as the metrics reporter, an HTTP endpoint starts with the context of `/metrics` in default port `9797` when starting the Ballerina service.
+
+## Step 3 - Configure Ballerina runtime configurations
+You can set up prometheus for your Ballerina project using configurations similar to the following in your `Config.toml` file.
+
+```toml
+[ballerina.observe]
+metricsEnabled=true
+metricsReporter="prometheus"
+
+[ballerinax.prometheus]
+port=9797
+host="0.0.0.0"
+```
+
+Configuration key | Description | Default value | Possible values 
+--- | --- | --- | --- 
+`ballerinax.prometheus.port` | The value of the port to which the '/metrics' service will bind. This service will be used by Prometheus to scrape the information of the Ballerina service. | `9797` | Any suitable value for port 0 - 0 - 65535. However, within that range, ports 0 - 1023 are generally reserved for specific purposes, therefore it is advisable to select a port without that range. 
+`ballerinax.prometheus.host` | The name of the host to which the '/metrics' service will bind. This service will be used by Prometheus to scrape the information of the Ballerina service. | `0.0.0.0` | IP or Hostname or 0.0.0.0 of the node in which the Ballerina service is running.
+
+## Step 4 - Run the Ballerina service
+
+When Ballerina observability is enabled, the Ballerina runtime exposes internal metrics via an HTTP endpoint (`/metrics`) for metrics monitoring and the metrics will be published to Prometheus. Prometheus should be configured to scrape metrics from the metrics HTTP endpoint in Ballerina.
+
+Run the following command to start the Ballerina service.
+
+```
+$ bal run
+
+Compiling source
+
+Running executable
+
+ballerina: started Prometheus HTTP listener 0.0.0.0:9797
+```
+
+## Step 5 - Send requests
+ 
+Send requests to <http://localhost:8090/shop/products>.
+
+Example cURL commands:
+
+```
+$ curl -X GET http://localhost:8090/shop/products
+```
+```
+$ curl -X POST http://localhost:8090/shop/product \
+-H "Content-Type: application/json" \
+-d '{
+    "id": 4, 
+    "name": "Laptop Charger", 
+    "price": 50.00
+}'
+```
+```
+$ curl -X POST http://localhost:8090/shop/order \
+-H "Content-Type: application/json" \
+-d '{
+    "productId": 1, 
+    "quantity": 1
+}'
+```
+```
+$ curl -X GET http://localhost:8090/shop/order/0
+```
+
+## Step 6 - View metrics on the Prometheus server
+
+Go to <http://localhost:19090/> and check whether you can see the Prometheus graph.
+Ballerina metrics should appear in the Prometheus graph's metrics list when the Ballerina service is started.
+
+![Ballerina metrics listed in Prometheus](/learn/images/ballerina-metrics-listed-in-prometheus.png "Ballerina metrics listed in Prometheus")
+
+![Ballerina metric in graph format](/learn/images/ballerina-metrics-in-graph.png "Ballerina metric in graph format")
+
+You can also use the following command to get the metrics.
+
+```
+$ curl http://localhost:9797/metrics
```

Comment:
This didn't give any results. instead gives connection error. Ballerina version is 12.3. please check 

<img width="1195" alt="image" src="https://github.com/user-attachments/assets/c9768937-87aa-4daf-aab2-d2563739d75c" />

Reply:
I tested with the sample provided with Ballerina `2201.12.4` and worked for me.
<img width="1260" alt="Screenshot 2025-06-04 at 15 54 50" src="https://github.com/user-attachments/assets/8122b7c6-42c5-4a83-b35a-2cdbc407b013" />

Reply:
Here you have put the runtime configurations in `Ballerina.toml` instead of `Config.toml`.

```toml
[ballerina.observe]
metricsEnabled=true
metricsReporter="prometheus"

[ballerinax.prometheus]
port=9797
host="0.0.0.0"
```